mne_icalabel/gui/_label_components.py

```diff
@@ -0,0 +1,274 @@
+from typing import Dict, List, Union
+
+from matplotlib import pyplot as plt
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+from mne import BaseEpochs
+from mne.io import BaseRaw
+from mne.preprocessing import ICA
+from mne.utils import _validate_type
+from mne.viz import set_browser_backend
+from qtpy.QtCore import Qt, Slot
+from qtpy.QtWidgets import (
+    QAbstractItemView,
+    QButtonGroup,
+    QGridLayout,
+    QListWidget,
+    QMainWindow,
+    QPushButton,
+    QVBoxLayout,
+    QWidget,
+)
+
+from mne_icalabel.config import ICLABEL_LABELS_TO_MNE
+
+
+class ICAComponentLabeler(QMainWindow):
+    """Qt GUI to annotate components.
+
+    Parameters
+    ----------
+    inst : Raw | Epochs
+    ica : ICA
+    """
+
+    def __init__(self, inst: Union[BaseRaw, BaseEpochs], ica: ICA, show: bool = True) -> None:
+        ICAComponentLabeler._check_inst_ica(inst, ica)
+        super().__init__()  # initialize the QMainwindow
+        set_browser_backend("qt")  # force MNE to use the QT Browser
+
+        # keep an internal pointer to the instance and to the ICA
+        self._inst = inst
```

Comment:
@adam2392 We could drop the EOG / ECG channels here. We don't need them anyway for this GUI.

Reply:
But this would modify the outside instance in place. Perhaps we leave it for now? It is kind of nice if you have it to compare against time-series traces that look "like it".

Reply:
As you want, we can always do a copy before dropping the channels.
It's True that it is a nice piece of information to have to compare against the IC's time-series. The main issue IMO is the size of that browser and especially of the scrollbar/statusbar.

Reply:
Let's just merge this for now for the sake of giving Aaron something to work with. We can modify as we go.

Reply:
Sounds good!